src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -443,7 +443,12 @@ else if (!usersInSameSlot.contains(username)) {
             int dateWeek = getWeekOfDate(date);
             int startDateWeek = getWeekOfDate(startDate);
             int weeksDifference;
-            weeksDifference = dateWeek < startDateWeek ? dateWeek == startDateWeek - 1 ? 0 : dateWeek + 53 - startDateWeek : dateWeek - startDateWeek;
+            // we need to take into account that years can have 53 or 52 calendar weeks
+            ZonedDateTime firstDateOfStartYear = ZonedDateTime.of(startDate.getYear(), 1, 1, 0, 0, 0, 0, startDate.getZone());
+            ZonedDateTime lastDateOfStartYear = ZonedDateTime.of(startDate.getYear(), 12, 31, 0, 0, 0, 0, startDate.getZone());
+            boolean beginsOrEndsOnThursday = firstDateOfStartYear.getDayOfWeek() == DayOfWeek.THURSDAY || lastDateOfStartYear.getDayOfWeek() == DayOfWeek.THURSDAY;
+            int lastCalendarWeek = beginsOrEndsOnThursday ? 53 : 52;
+            weeksDifference = dateWeek < startDateWeek ? dateWeek == startDateWeek - 1 ? 0 : dateWeek + lastCalendarWeek - startDateWeek : dateWeek - startDateWeek;
```

Comment:
```suggestion
            int weeksDifference = dateWeek < startDateWeek ? dateWeek == startDateWeek - 1 ? 0 : dateWeek + lastCalendarWeek - startDateWeek : dateWeek - startDateWeek;
```
With the double ternary operator it is also quite hard to understand what is going on in this line. Can you change it into a nested if-else, please?